CommandLine/emake-tests/Parsing/parser-test.cpp

```diff
@@ -35,7 +35,244 @@ struct ParserTester {
 
 void assert_identifier_is(AST::Node *node, std::string_view name) {
   ASSERT_EQ(node->type, AST::NodeType::IDENTIFIER);
-  ASSERT_EQ(dynamic_cast<AST::IdentifierAccess *>(node)->name.content, name);
+  ASSERT_EQ(node->As<AST::IdentifierAccess>()->name.content, name);
+}
+
+std::string ExpectedMsg = "";
+
+MATCHER_P(IsDeclaration, decls, "") {
+  bool b1 = arg->type == AST::NodeType::DECLARATION;
+  if (!b1) {
```

Comment:
For this and all remaining matchers, this check will be the first one before the casting to ensure that we do not encounter a `Segmentation fault`.